File: OnlineStore/OnlineStoreUI/src/model/entities/apiRequests/productSearchRequest.ts
export class ProductSearchRequest{
    minCost: number;
    maxCost: number;
    productSearch: string;
    productStatuses:string[];
    idProductCategories: number[];

    constructor(minCost: number,maxCost: 
        number,productSearch: string, 
        productStatuses:string[],
        idProductCategories: number[]){

        this.minCost = minCost;
        this.maxCost = maxCost;
        this.productSearch = productSearch;
        this.productStatuses = productStatuses;
        this.idProductCategories = idProductCategories;
    }
}

File: OnlineStore/Api/BLContracts/Models/ProductSearchRequest.cs
using System.Collections.Generic;

namespace BLContracts.Models
{
	public class ProductSearchRequest
	{
		public decimal MinCost { get; set; }
		public decimal MaxCost { get; set; }
		public string ProductSearch { get; set; }
		public IEnumerable<string> ProductStatuses { get; set; }
		public IEnumerable<int> IdProductCategories { get; set; }
	}
}
